src/field/field255.rs

```diff
@@ -0,0 +1,487 @@
+// Copyright (c) 2023 ISRG
+// SPDX-License-Identifier: MPL-2.0
+
+//! Finite field arithmetic for `GF(2^255 - 19)`.
+
+use crate::{
+    codec::{CodecError, Decode, Encode},
+    field::{FieldElement, FieldElementVisitor, FieldError},
+};
+use fiat_crypto::curve25519_64::{
+    fiat_25519_add, fiat_25519_carry, fiat_25519_carry_mul, fiat_25519_from_bytes, fiat_25519_opp,
+    fiat_25519_sub, fiat_25519_tight_field_element, fiat_25519_to_bytes,
+};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    convert::TryFrom,
+    fmt::{self, Debug, Display, Formatter},
+    io::{Cursor, Read},
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+};
+use subtle::{Choice, ConstantTimeEq, ConstantTimeGreater, ConstantTimeLess};
+
+#[cfg(test)]
+pub(crate) mod integer;
+
+#[cfg(test)]
+use integer::Field255Integer;
+
+// `python3 -c "print(', '.join(hex(x) for x in (2**255-19).to_bytes(32, 'big')))"`
+const MODULUS_BIG_ENDIAN: [u8; 32] = [
+    0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xed,
+];
+
+/// `GF(2^255 - 19)`, a 255-bit field.
+#[derive(Clone, Copy)]
+pub struct Field255(fiat_25519_tight_field_element);
+
+impl Field255 {
+    /// Attempts to instantiate a `Field255` from the first `Self::ENCODED_SIZE` bytes in the
+    /// provided slice.
+    ///
+    /// # Errors
+    ///
+    /// An error is returned if the provided slice is not long enough to encode a field element or
+    /// if the decoded value is greater than the field prime.
+    fn try_from_bytes(bytes: &[u8], mask_top_bit: bool) -> Result<Self, FieldError> {
+        if Self::ENCODED_SIZE > bytes.len() {
+            return Err(FieldError::ShortRead);
+        }
+
+        let mut value = [0u8; Self::ENCODED_SIZE];
+        value.copy_from_slice(&bytes[..Self::ENCODED_SIZE]);
+
+        if mask_top_bit {
+            value[0] &= 0b0111_1111;
+        }
+
```

Comment:
Nice work :) We should file an issue to do the same thing for the other fields. (We are less careful with checking for modulus overflow there.)

Reply:
Disregard, I'll send a follow-up PR